src/core/app.lua

```diff
@@ -130,27 +130,63 @@ function configure (new_config)
    counter.add(configs)
 end
 
+-- Removes the claim on a name, freeing it for other programs.
+--
+-- This relinquish a claim on a name if one exists. if the name does not
+-- exist it will raise an error with an error message.
+function unclaim_name(claimed_name)
+   local name = assert(claimed_name or configuration.name, "No claim to name.")
+   local name_fq = named_program_root .. "/" .. name
+   local piddir = assert(S.readlink(name_fq))
+   local backlink = piddir .. "/name"
+
+   -- First unlink the backlink
+   assert(S.unlink(backlink))
+
+   -- Remove the actual namedir
+   assert(S.unlink(name_fq))
+
+   -- Remove from the name from the configuration
+   configuration.name = nil
```

Comment:
We _could_ but it depends what we want out of it being a global variable. If the hope is that we have a global `claimed_name` variable so that we don't have to pass it to `claim_name` and `unclaim_name` then, no I don't think that'd work. There are a few reasons:

1. When claming a new name, we need to know the old to unclaim it
2. When claming a new name, we need to know the old so we can fall back to it, if something were to go wrong.
3. Maybe not a huge issue but if it were changed by others (i.e. not by `claim_name` or `unclaim_name`) then I couldn't reliably do [this check](https://github.com/Igalia/snabb/pull/742/files#diff-e2bf611b7cf61fb539b42bf778b8fc9cR170).

On the other hand, if we want to just use a global variable... just because. Then sure, provided we only change it inside claim_name.